### PR TITLE
SS-2168 - User profile warnings

### PR DESCRIPTION
**Pull request by AL-Session**
### Contributor checklist

- [X] I have tested my contribution on these devices:
 * Virtual Pixel 3a, Android 9 API 28
- [X] My contribution is fully baked and ready to be merged as is
~~- [ ] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)~~

----------

### Description
This PR is to add user feedback in the following situations:
- Profile picture upload failed (Toast: "Failed to update profile"), and
- Profile picture removal failed (Toast: "Failed to remove display picture")

The OkHttp, conscrypt, and androidx core libraries were updated in the process of experimentation to add this functionality. Although the final result can be achieved without these updates, I see little reason to roll them back to our old, stale library versions when these fresh versions work just fine.

To test this PR you might like to:
- Put the device in Airplane mode and attempt to upload a profile picture - when it fails you'll see the toast "Failed to update profile", and
- With an existing profile picture, put the device in Airplane mode and then attempt to remove the profile picture. When the network sync of the profile picture removal fails the error toast "Failed to remove profile picture" will display.

Note: while we _could_ remove the local profile picture even if the network sync failed, this would be confusing and would also deny the user the chance to retry removal. Also, it could look **to them** as if they had no profile picture, but online to others they could - so it's best not to break that sync imho.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -33,7 +33,7 @@ kovenantVersion=3.3.0
 lifecycleVersion=2.5.1
 materialVersion=1.8.0
 mockitoKotlinVersion=4.1.0
-okhttpVersion=3.12.1
+okhttpVersion=5.0.0-alpha.14
```
Comment (simophin) [inline gradle.properties:36]: Hey is there any reason to use alpha version? Can we opt for a stable version instead?

File: app/src/main/java/org/thoughtcrime/securesms/preferences/SettingsActivity.kt [hunk 2]
```diff
@@ -233,21 +234,39 @@ class SettingsActivity : PassphraseRequiredActionBarActivity() {
...
             TextSecurePreferences.setProfileName(this, displayName)
             configFactory.user?.setName(displayName)
         }
+
         val encodedProfileKey = ProfileKeyUtil.generateEncodedProfileKey(this)
+
+        // If we're updating the profile picture..
+        // Note: We define the promise here so we have the scope to check success/fail later without
+        // having to nest the blocks inside the conditional blocks.
+        lateinit var uploadProfilePicturePromise: Promise<*, Exception>
```
[the remark this answers is not in the document]
Comment (AL-Session) [reply app/src/main/java/org/thoughtcrime/securesms/preferences/SettingsActivity.kt:249]: Nice - done!

File: gradle.properties (hunk 1 above)
Comment (AL-Session) [reply gradle.properties]: Rolled back to stable `4.12.0` version of okhttp

**Review by simophin — APPROVED** [review]

**Comment by AL-Session** [comment]
Reverted previous `dev` merge and force-pushed updated `dev` merge.

**Pull request merged**